### PR TITLE
update to v0.9.40

### PR DESCRIPTION
**Pull request by nbaztec**
updates substrate dependencies to v0.9.40

### CODE REVIEW

File: rust-toolchain [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [toolchain]
-channel = "nightly-2022-11-15"
+channel = "stable"
```
Comment (JoshOrndorff) [inline rust-toolchain:2]: :exploding_head:

File: parachain-template/runtime/src/lib.rs [hunk 2]
```diff
@@ -752,15 +764,15 @@ impl_runtime_apis! {
 		) {
 			use frame_benchmarking::{list_benchmark, Benchmarking, BenchmarkList};
 			use frame_support::traits::StorageInfoTrait;
-			use frame_system_benchmarking::Pallet as SystemBench;
-			use cumulus_pallet_session_benchmarking::Pallet as SessionBench;
+			// use frame_system_benchmarking::Pallet as SystemBench;
```
Comment (nbaztec) [inline parachain-template/runtime/src/lib.rs:767]: @librelois I was unable to make these benchmarks work with `--features runtime-benchmarks`
`SystemBench`, `CollatorSelection` complains of unable to guess types
`cumulus_pallet_session_benchmarking` requires `pallet_session::Config` to be implemented which leads to other errors.

Any help is appreciated

**Comment by girazoki** [comment]
I see that in master we point to upstream repos, and not moonbeam forks, is that right? I think that makes it easier for other projects to use nimbus

**Comment by nbaztec** [comment]
Closing in favor of https://github.com/PureStake/nimbus/pull/91

**Pull request closed without merging**